processor/filterprocessor/metrics.go

```diff
@@ -117,121 +106,75 @@ func newFilterMetricProcessor(logger *zap.Logger, cfg *Config) (*filterMetricPro
-	filteringDataPoints := fmp.dataPointConditions != nil
-
-	if filteringMetrics || filteringDataPoints {
-		var errors error
-		pdm.ResourceMetrics().RemoveIf(func(rmetrics pmetric.ResourceMetrics) bool {
-			rmetrics.ScopeMetrics().RemoveIf(func(smetrics pmetric.ScopeMetrics) bool {
-				smetrics.Metrics().RemoveIf(func(metric pmetric.Metric) bool {
-					if filteringMetrics {
-						tCtx := ottlmetric.NewTransformContext(metric, smetrics.Scope(), rmetrics.Resource())
-						metCondition, err := common.CheckConditions(ctx, tCtx, fmp.metricConditions)
-						if err != nil {
-							errors = multierr.Append(errors, err)
-						}
-						if metCondition {
-							return true
-						}
-					}
-					if filteringDataPoints {
-						switch metric.Type() {
-						case pmetric.MetricTypeSum:
-							err := fmp.handleNumberDataPoints(ctx, metric.Sum().DataPoints(), metric, smetrics.Metrics(), smetrics.Scope(), rmetrics.Resource())
-							if err != nil {
-								errors = multierr.Append(errors, err)
-							}
-							return metric.Sum().DataPoints().Len() == 0
-						case pmetric.MetricTypeGauge:
-							err := fmp.handleNumberDataPoints(ctx, metric.Gauge().DataPoints(), metric, smetrics.Metrics(), smetrics.Scope(), rmetrics.Resource())
-							if err != nil {
-								errors = multierr.Append(errors, err)
-							}
-							return metric.Gauge().DataPoints().Len() == 0
-						case pmetric.MetricTypeHistogram:
-							err := fmp.handleHistogramDataPoints(ctx, metric.Histogram().DataPoints(), metric, smetrics.Metrics(), smetrics.Scope(), rmetrics.Resource())
-							if err != nil {
-								errors = multierr.Append(errors, err)
-							}
-							return metric.Histogram().DataPoints().Len() == 0
-						case pmetric.MetricTypeExponentialHistogram:
-							err := fmp.handleExponetialHistogramDataPoints(ctx, metric.ExponentialHistogram().DataPoints(), metric, smetrics.Metrics(), smetrics.Scope(), rmetrics.Resource())
-							if err != nil {
-								errors = multierr.Append(errors, err)
-							}
-							return metric.ExponentialHistogram().DataPoints().Len() == 0
-						case pmetric.MetricTypeSummary:
-							err := fmp.handleSummaryDataPoints(ctx, metric.Summary().DataPoints(), metric, smetrics.Metrics(), smetrics.Scope(), rmetrics.Resource())
-							if err != nil {
-								errors = multierr.Append(errors, err)
-							}
-							return metric.Summary().DataPoints().Len() == 0
-						default:
-							return false
-						}
-					}
-					return false
-				})
-				return smetrics.Metrics().Len() == 0
-			})
-			return rmetrics.ScopeMetrics().Len() == 0
-		})
-
-		if errors != nil {
-			return pdm, errors
-		}
-		if pdm.ResourceMetrics().Len() == 0 {
-			return pdm, processorhelper.ErrSkipProcessingData
-		}
-		return pdm, nil
+func (fmp *filterMetricProcessor) processMetrics(ctx context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
+	if fmp.skipResourceExpr == nil && fmp.skipMetricExpr == nil && fmp.skipDataPointExpr == nil {
```

Comment:
In the next PR, I will investigate how to remove the need of the "skipResourceExpr". This is here because that's how the old code was implemented, I believe we can incorporate in the "skipMetric" but I need to check carefully that. For the moment I keep it as this.